backend/src/main/java/quarano/security/web/AuthenticationController.java

```diff
@@ -60,17 +68,28 @@
 						.ifPresent(people::save))
 				.<HttpEntity<?>> map(accountRepresentations::toTokenResponse)
 				.recover(EmptyResultDataAccessException.class, it -> toUnauthorized(it.getMessage()))
+				.recover(ExternalZipException.class, it -> toUnprocessable(it.getMessage()))
```

Comment:
Let's stick with `toUnauthorized(…)`. If at some point we need the client to properly distinguish reasons for failed logins, we'd rather have to introduce proper domain error codes. For now it should be sufficient to just show the error message.

Reply:
We should give the user useful feedback and not "username or password is wrong" when we know it better.

Reply:
That's not what I was suggesting. Which status code we return and which error message we return is orthogonal topics. It's semantically wrong to claim we cannot process the entity. We reject authentication based on the given credentials. We can still do so giving different rationales. And we already actually do that for closed cases and invalid passwords in general.

Reply:
Wouldn't 403 "Forbidden" be the right status? I noticed this while reading the description: https://developer.mozilla.org/en/docs/Web/HTTP/Status/401
I can't see a message for closed cases in the response. I think, there is a bug.

Reply:
Yeah, one could argue 403 because the login is generally successful (password matches) but the user is lacking permissions, i.e. is not authorized to use the application.

I was just judging from the different exception messages piped into `toUnauthorized(…)` to forward them into the response body. I guess we need to i18n those properly to let the client use the reasonably.

Reply:
It is 403 now and I have improved the processing of the messages in the response and their translation.